billing/src/main/java/org/stepik/android/view/injection/billing/BillingModule.kt

```diff
@@ -1,37 +1,54 @@
 package org.stepik.android.view.injection.billing
 
 import android.content.Context
+import android.util.Log
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.BillingClientStateListener
+import com.android.billingclient.api.BillingResult
+import com.android.billingclient.api.Purchase
+import com.jakewharton.rxrelay2.BehaviorRelay
 import dagger.Module
 import dagger.Provides
 import io.reactivex.Scheduler
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
-import org.solovyev.android.checkout.Billing
-import org.solovyev.android.checkout.Checkout
 import ru.nobird.android.view.injection.base.RxScheduler
 
 @Module
 class BillingModule {
-    @Provides
-    internal fun provideBilling(
-        context: Context,
-        @PublicLicenseKey
-        appPublicLicenseKey: String
-    ): Billing =
-        Billing(context, object : Billing.DefaultConfiguration() {
-            override fun getPublicKey(): String =
-                appPublicLicenseKey
-        })
-
     /**
      * Provides system checkout that can be used for querying inventory & etc.
      */
     @Provides
-    @SystemCheckout
-    internal fun provideSystemCheckout(billing: Billing): Checkout =
-        Checkout
-            .forApplication(billing)
-            .also(Checkout::start)
+    @BillingSingleton
+    internal fun provideBillingClient(
+        context: Context,
+        purchaseListenerBehaviorRelay: BehaviorRelay<Pair<BillingResult, List<Purchase>?>>
+    ): BillingClient =
+        BillingClient
+            .newBuilder(context)
+            .setListener { billingResult, mutableList ->
+                Log.d("APPS", "Result: ${billingResult.debugMessage} Code: ${billingResult.responseCode}")
+                purchaseListenerBehaviorRelay.accept(billingResult to mutableList)
+            }
+            .enablePendingPurchases()
```

Comment:
will we receive delayed payments here?

Reply:
Yes, this listener will receive purchases with a _Pending_ status